src/v/cluster/controller_backend.cc

```diff
@@ -389,23 +412,29 @@ controller_backend::execute_partitition_op(const topic_table::delta& delta) {
     case op_t::del:
         return delete_partition(delta.ntp, rev);
     case op_t::update:
-        return process_partition_update(delta.ntp, delta.new_assignment, rev);
+        vassert(
```

Comment:
Maybe we should rename `vassert` to `crash_if` so we are more motivated to add as much context as possible